gnmi/gnmiclient/cache.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gnmiclient
+
+import (
+	"context"
+	"io"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/peer"
+
+	gpb "github.com/openconfig/gnmi/proto/gnmi"
+)
+
+// NewCacheClient creates a gNMI client for the gNMI cache.
+// The client calls the server gRPC implementation directly to reduce overhead.
```

Comment:
The client calls the server gRPC implementation with a custom streaming gRPC implementation in order to bypass the regular gRPC wire marshalling/unmarshalling handling.